### PR TITLE
Throw UnsupportedConstraint error in copy_to

### LINKED ISSUE #79: KeyError: key MathOptInterface.VariableIndex(1) not found
```julia
import MathOptInterface
const MOI = MathOptInterface
import JuMP
import CSDP
```
```julia
m = JuMP.Model(CSDP.Optimizer)
JuMP.@variable(m, x[1:2] >= 0)
JuMP.@constraint(m, x[1] + 2*x[2] == 1.0)
JuMP.@objective(m, Min, 3*x[1]^2 + 4*x[2]^2 + 2*x[1]*x[2] + 5*x[1] + x[2])
JuMP.optimize!(m)
```

```julia
julia> print(m.moi_backend)
Minimize ScalarQuadraticFunction{Float64}:
 0.0 + 5.0 x[1] + 1.0 x[2] + 3.0 x[1]² + 4.0 x[2]² + 2.0 x[2]*x[1]

Subject to:

ScalarAffineFunction{Float64}-in-EqualTo{Float64}
 0.0 + 1.0 x[1] + 2.0 x[2] == 1.0

VariableIndex-in-GreaterThan{Float64}
 x[1] >= 0.0
 x[2] >= 0.0
```

```julia
m_ = MOI.Utilities.CachingOptimizer(
    MOI.Utilities.Model{Float64}(),
    CSDP.Optimizer(),
)
MOI.Utilities.reset_optimizer(m_)
MOI.Utilities.attach_optimizer(m_)
x = MOI.add_variables(m_, 2)
MOI.add_constraint(m_, x[1], MOI.GreaterThan(0.0))
MOI.add_constraint(m_, x[2], MOI.GreaterThan(0.0))
MOI.add_constraint(m_, 
    MOI.ScalarAffineFunction([MOI.ScalarAffineTerm(1.0, x[1]), MOI.ScalarAffineTerm(2.0, x[2])], 0.0), 
    MOI.EqualTo(1.0)
)
MOI.set(m_, MOI.ObjectiveFunction{MOI.ScalarQuadraticFunction{Float64}}(), 
    MOI.ScalarQuadraticFunction(
        [MOI.ScalarQuadraticTerm(6.0, x[1], x[1]), MOI.ScalarQuadraticTerm(8.0, x[2], x[2]), MOI.ScalarQuadraticTerm(2.0, x[1], x[2])], 
        [MOI.ScalarAffineTerm(5.0, x[1]), MOI.ScalarAffineTerm(1.0, x[2])], 
        0.0,
    )
)
MOI.set(m_, MOI.ObjectiveSense(), MOI.MIN_SENSE)
MOI.optimize!(m_)

ERROR: KeyError: key MathOptInterface.VariableIndex(1) not found
Stacktrace:
 [1] getindex
   @ .julia\packages\MathOptInterface\Ohzb2\src\Utilities\CleverDicts.jl:171 [inlined]
 [2] getindex
   @ .julia\packages\MathOptInterface\Ohzb2\src\Utilities\copy\index_map.jl:64 [inlined]
 [3] copy_to(dest::CSDP.Optimizer, src::MathOptInterface.Utilities.Model{Float64})
   @ CSDP .julia\packages\CSDP\4LfHL\src\MOI_wrapper.jl:300
 [4] optimize!
   @ .julia\packages\MathOptInterface\Ohzb2\src\MathOptInterface.jl:86 [inlined]
 [5] optimize!(m::MathOptInterface.Utilities.CachingOptimizer{CSDP.Optimizer, MathOptInterface.Utilities.Model{Float64}})
   @ MathOptInterface.Utilities .julia\packages\MathOptInterface\Ohzb2\src\Utilities\cachingoptimizer.jl:316
```
```julia
julia> print(m_)
Minimize ScalarQuadraticFunction{Float64}:
 0.0 + 5.0 v[1] + 1.0 v[2] + 3.0 v[1]² + 4.0 v[2]² + 2.0 v[1]*v[2]

Subject to:

ScalarAffineFunction{Float64}-in-EqualTo{Float64}
 0.0 + 1.0 v[1] + 2.0 v[2] == 1.0

VariableIndex-in-GreaterThan{Float64}
 v[1] >= 0.0
 v[2] >= 0.0
```
I don't know what is wrong here, the MOI model seems to be the same.

### PR DESCRIPTION
**Pull request by odow**
Closes #79

### CODE REVIEW

File: src/MOI_wrapper.jl [hunk 1]
```diff
@@ -274,7 +284,7 @@ function MOI.copy_to(dest::Optimizer, src::MOI.ModelLike)
         MOI.PositiveSemidefiniteConeTriangle,
     )
     vis_src = MOI.get(src, MOI.ListOfVariableIndices())
-    if length(vis_src) < length(index_map.var_map)
+    if length(vis_src) != length(index_map.var_map)
```
Comment (odow) [inline src/MOI_wrapper.jl]: Ooops. I guess this should be a separate PR. Is the current `<` correct? Shouldn't it be the other way around (or `!=`)?

File: test/MOI_wrapper.jl [hunk 2]
```diff
@@ -36,6 +36,18 @@ function test_options()
     )
 end
 
+function test_unsupported_constraint()
+    model = MOI.Utilities.Model{Float64}()
+    x = MOI.add_variable(model)
+    MOI.add_constraint(model, x, MOI.GreaterThan(0.0))
+    MOI.add_constraint(model, 1.0 * x, MOI.EqualTo(0.0))
+    @test_throws(
+        MOI.UsupportedConstraint{MOI.VariableIndex,MOI.GreaterThan{Float64}}(),
```
Comment (odow) [inline test/MOI_wrapper.jl]:
```suggestion
        MOI.UnsupportedConstraint{MOI.VariableIndex,MOI.GreaterThan{Float64}}(),
```

**Review by blegat — APPROVED** [review]
This should be mirrored to SDPA. It would be good to have this tested with `runtests`, I thought it was the case

**Comment by odow** [comment]
> It would be good to have this tested with runtests, I thought it was the case

I think it is. But CSDP skips a lot of the tests.

**Pull request merged**